libphx/src/str_map.rs

```diff
@@ -28,7 +28,7 @@ pub struct StrMapIter {
 
 #[inline]
 unsafe extern "C" fn Hash(key: *const libc::c_char) -> u64 {
-    Hash_XX64(key as *const _, StrLen(key) as i32, 0)
+    Hash_XX64(key as *const _, key.convert().len() as i32, 0)
```

Comment:
This is far less efficient than before, it's going to copy everything we try to hash at least once now.

If we enhance the trait as I mentioned in an earlier comment, we could turn this into `key.to_slice().len()` which would be as fast as `StrLen(..)` was.

Reply:
Now convert is the same as your StrToSlice